deckhouse-controller/pkg/apis/deckhouse.io/validation/validate_cluster_configuration.go

```diff
@@ -114,35 +115,55 @@ func validateDefaultCRI(defaultCRI string, cli client.Client) (*kwhvalidating.Va
 	}
 }
 
-func clusterConfigurationHandler(mm moduleManager, cli client.Client) http.Handler {
-	validator := kwhvalidating.ValidatorFunc(func(ctx context.Context, ar *model.AdmissionReview, obj metav1.Object) (*kwhvalidating.ValidatorResult, error) {
-		secret, ok := obj.(*v1.Secret)
-		if !ok {
-			log.Debug("unexpected type", log.Type("expected", v1.Secret{}), log.Type("got", obj))
-			return nil, fmt.Errorf("expect Secret as unstructured, got %T", obj)
-		}
+func validateClusterConfiguration(schemaStore *config.SchemaStore, clusterConfiguration []byte) (*kwhvalidating.ValidatorResult, error) {
+	_, err := schemaStore.Validate(&clusterConfiguration, config.ValidateOptionOmitDocInError(true))
+	if err != nil {
+		return rejectResult(err.Error())
+	}
 
-		clusterConfigurationRaw, ok := secret.Data["cluster-configuration.yaml"]
-		if !ok {
-			log.Debug("no cluster-configuration found in secret", slog.String("namespace", obj.GetNamespace()), slog.String("name", obj.GetName()))
-			return nil, fmt.Errorf("expected field 'cluster-configuration.yaml' not found in secret %s", secret.Name)
-		}
+	return allowResult(nil)
+}
 
-		clusterConf := new(clusterConfig)
-		if err := yaml.Unmarshal(clusterConfigurationRaw, clusterConf); err != nil {
-			log.Debug("failed to unmarshal cluster configuration", log.Err(err))
-			return nil, fmt.Errorf("unmarshal cluster configuration: %w", err)
+func clusterConfigurationHandler(mm moduleManager, cli client.Client, schemaStore *config.SchemaStore) http.Handler {
+	validator := kwhvalidating.ValidatorFunc(func(ctx context.Context, ar *model.AdmissionReview, obj metav1.Object) (*kwhvalidating.ValidatorResult, error) {
+		switch ar.Operation {
```

Comment:
looks like a more clear handling
```suggestion
		if ar.Operation == model.OperationDelete {
```